### PR TITLE
samples: sensor: bq274xx: Read Gauge Channel Individually.

### LINKED ISSUE #28383: bq274xx sample is not working
**Describe the bug**
I am trying to get bq27421 up and running on our developement board(soc:nrf52840).But sample is not working, because in sample trying to fetch samples for all channels(SENSOR_CHAN_ALL) ,but driver does not handling SENSOR_CHAN_ALL
**To Reproduce**
1. west build 
2. Opened minicom
4. Seeing values as zero for all the channels

**Environment (please complete the following information):**
 - OS: Linux
 - gcc-arm-none-eabi-9-2020-q2-update
 - revision: 0742a45ac9f51d45f7b1fe148ac7d9bbfecb2f83

### PR DESCRIPTION
**Pull request by Navin-Sankar**
Read Gauge Channel Individually.
Fixes #28383.

Signed-off-by: NavinSankar Velliangiri <navin@linumiz.com>

### CODE REVIEW

**Comment by Navin-Sankar** [comment]
@jrananth1032 we don't have hardware. Test the sample at you end and confirm

**Comment by jrananth1032** [comment]
sure , i will test and comfirm

**Comment by jrananth1032** [comment]
> @jrananth1032 we don't have hardware. Test the sample at you end and confirm

It is working perfectly

**Review by MaureenHelm — APPROVED** [review]
thanks @Navin-Sankar

File: samples/sensor/bq274xx/src/main.c [hunk 1]
```diff
@@ -30,95 +30,172 @@ static void do_main(const struct device *dev)
 		full_charge_capacity, remaining_charge_capacity, avg_power,
 		int_temp, current_standby, current_max_load, state_of_health;
 
-	int status = 0;
+	int ret = 0;
 
 	while (1) {
-		status = sensor_sample_fetch(dev);
-		if (status < 0) {
+		ret = sensor_sample_fetch_chan(dev, SENSOR_CHAN_GAUGE_VOLTAGE);
```
Comment (gmarull) [inline samples/sensor/bq274xx/src/main.c:36]: This is the only change required to actually fix the bug, so other changes should be put into another commit as they are just name changes and error flow changes.

**Review by gmarull — CHANGES_REQUESTED** [review]

**Review by gmarull — CHANGES_REQUESTED** [review]
Please, check commits messages and content, doesn't seem to be right.

**Comment by Navin-Sankar** [comment]
> Please, check commits messages and content, doesn't seem to be right.

changed the commit message.

File: samples/sensor/bq274xx/src/main.c [hunk 2]
```diff
@@ -26,14 +26,13 @@ static void bq274xx_show_values(const char *type, struct sensor_value value)
...
+	int ret = 0;
 	struct sensor_value voltage, current, state_of_charge,
 		full_charge_capacity, remaining_charge_capacity, avg_power,
 		int_temp, current_standby, current_max_load, state_of_health;
 
-	int status = 0;
-
 	while (1) {
-		status = sensor_sample_fetch(dev);
+		ret = sensor_sample_fetch_chan(dev, SENSOR_CHAN_GAUGE_VOLTAGE);
```
Comment (gmarull) [inline samples/sensor/bq274xx/src/main.c:35]: In this commit next statement does not check for the returned code.

**Review by gmarull — CHANGES_REQUESTED** [review]
There's an issue in the first commit, second commit message does not align with the changes, which are just a variable name change.

**Comment by Navin-Sankar** [comment]
@gmarull @MaureenHelm ping.

**Review by MaureenHelm — COMMENTED** [review]
@Navin-Sankar please rebase

@gmarull please revisit

**Comment by Navin-Sankar** [comment]
@MaureenHelm @gmarull patch rebased.

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**